create_db.py

```diff
@@ -0,0 +1,265 @@
+	return seed
+
+# create a pickle file that contains spacy object of every line in the corpus
+def create_corpus():
+	nlp = spacy.load("de_core_news_sm")
+	text = []
+	with open('files/train.txt') as train:
+	    for line in train:
+	    	text.append(nlp(line))
+
+	with open('files/corpus.pkl', 'wb') as handle:
+	    pickle.dump(text, handle)
+
+def load_corpus():
+	obj = open('files/corpus.pkl', 'rb')
+	corpus = pickle.load(obj)
+	obj.close()
+	return corpus
+
+def get_corpus():
+	if os.path.exists("files/corpus.pkl"):
+		corpus = load_corpus()
+	else:
+		create_corpus()
+		corpus = load_corpus()
+	return corpus
+
+def rule_based_classification(corpus, seed):
+	count = 0
+	correct = 0
+	for line in corpus:
+	    for idx, token in enumerate(line):
+	    	if token.pos_ == "NOUN":
+	    		# get pronoun for current token
+		    	pronoun = (line[idx - 1]).text
+		    	# get case of current token
+		    	case = token.morph.get("Case")
+		    	# get text of token
+		    	text = token.text
+		    	# get gold data for gender, to compare against the calculated gender
+		    	spacy_gender = token.morph.get("Gender")
+		    	# assign gender to None
+		    	gender = None
+
+		    	# german grammar rules for pronoun/nounsS
+	    		if pronoun in ["der", "die", "das", "dem", "des", "den"]:
+		    		if pronoun == "der":
+		    			if case == "Nom":
+		    				gender = "Masc"
+		    			elif case == "Dat" or case == "Gen":
+		    				gender = "Fem"
+		    		elif pronoun == "die":
+		    			gender = "Fem"
+		    		elif pronoun == "das":
+		    			gender = "Neut"
+		    		elif pronoun == "dem" or pronoun == "des":
+		    			gender = "Masc"
+		    		elif pronoun == "den" and case == "Acc":
+		    			gender = "Masc"
+
+			    	if gender and spacy_gender:
+			    		count = count + 1
+			    		if gender == spacy_gender[0]:
+			    			correct = correct + 1
+			    			seed[gender].add(text)
+
+	print(f'Total nouns where a gender was assigned and Spacy has the gender categorised is {count}')
+	print(f'Total number of nouns correctly labelled (considering spacy gender categorisation as gold data)is {correct}. Hence, accuracy is {(correct/count)*100}')
+	return seed
+
+
+def create_suffix_bank(seed):
+	suffixes_masc = {}
+	suffixes_fem = {}
+	suffixes_neut = {}
+
+	# Masculine
+	c = 0
+	# loop through keys and create suffixes for each noun. only add suffixes of length less than or equal to 3
+	for nouns in set(seed["Masc"]):
+		suffixes = [nouns[i:] for i in range(len(nouns) - 3, len(nouns))]
+		for s in suffixes:
+			if s in suffixes_masc.keys():
+				suffixes_masc[s] = suffixes_masc[s] + 1
+			else:
+				suffixes_masc[s] = 1
+
+	# delete suffixes with frequency less than 3. Generate probability distribution for rest of the suffixes
+	for key in list(suffixes_masc.keys()):
+		if suffixes_masc[key] < 3:
+			del suffixes_masc[key]
+		else:
+			c = c + suffixes_masc[key]
+
+	for key, val in suffixes_masc.items():
+		suffixes_masc[key] = suffixes_masc[key]/c
```

Comment:
```suggestion
	suffixes_masc = {key: val/c for key, val in suffixes_masc.items() }
```
(btw why do you write `suffixes_masc[key]/c` when you have `val` so `val/c` is enough)

Reply:
agreed and fixed. using list comprehensions and counter to calculate